SdlLogs/SdlLogs.py

```diff
@@ -3,157 +3,289 @@
+import shutil
+import yaml
+
+
+class SettingsTags:
+    """
+    Contains useful attributes from settings file.
+
+    Attributes
+    ----------
+    NAME : const str
+        Name of plugin.
+    EXTENTION : const str
+        Extention of plugin settings file.
+    SOURCE_PATH : const str
+        Name settings attribute.
+    """
+    NAME = "SdlLogs"
+    EXTENTION = ".sublime-settings"
+    SOURCE_PATH = "source_path"
+
+
+class LogSyntax:
+    """
+    Contains regular expretion which parsed from yaml file.
+
+    Attributes
+    ----------
+    NAME : const str
+        Name of syntax attribute in settings file.
+    Tags : class Tags
+        Contains name of attributes in yaml file.
+
+    Methods
+    -------
+    load_syntax(self, pathToSettings)
+        Load regular expression from yaml file.
+
+    """
+    NAME = "syntax"
+
+    class Tags:
+        """
+        Contains name of attributes in yaml file.        
+        """
+        VARIABLES = "variables"
+        DATE_TIME = "date_time"
+        ID = "id"
+        COMPONET = "component"
+        PATH = "path"
+        LINE = "line"
+        MESSAGE = "message"
+        JUNK = "junk"
+        THREAD_ENTER = "thread_enter"
+        THREAD_EXIT = "thread_exit"
+
+    def __init__(self):
+        self.date_time = ""
+        self.id = ""
+        self.component = ""
+        self.path = ""
+        self.line = ""
+        self.message = ""
+        self.junk = ""
+        self.thread_enter = ""
+        self.thread_exit = ""
+
+    def load_syntax(self, pathToSettings):
+        """
+        Load regular expression from yaml file.
+        Search pattern for load syntax:
+            Order regular expression in yaml file should as:
+                variables:
+                    name_regex: some_regex
+                    ...
+        Args:
+            pathToSettings : str
+                path to setting file
+        """
+        obj = sublime.load_settings(pathToSettings).get(self.NAME)
+        print(obj)
+        if not obj:
+            sublime.error_message("Absent path to syntax file")
+        else:
+            yaml_file = yaml.load(sublime.load_resource(obj))
+            regex = yaml_file[self.Tags.VARIABLES]
+            self.date_time = regex[self.Tags.DATE_TIME]
+            self.id = regex[self.Tags.ID]
+            self.component = regex[self.Tags.COMPONET]
+            self.path = regex[self.Tags.PATH]
+            self.line = regex[self.Tags.LINE]
+            self.message = regex[self.Tags.MESSAGE]
+            self.junk = regex[self.Tags.JUNK]
+            self.thread_enter = regex[self.Tags.THREAD_ENTER]
+            self.thread_exit = regex[self.Tags.THREAD_EXIT]
+
+
+def get_selected_text(self):
+    """
+    Return selected text from view.
+
+    Parameters
+    ----------
+    self : class view 
+
+    Returns
+    ----------
+    str
+        Selected text
+    """
+    sel_text = self.view.sel()
+    sel_text = self.view.substr(sel_text[0])
```

Comment:
@dkandyba check if sel_text is empty

Reply:
@LuxoftAKutsan not necessary check it. This works when cursor focus on view and if text is not selected will return empty string